modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -242,6 +243,13 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
 
   private def validateHandshake(registrationRequest: RegistrationRequest, remoteAddress: Option[Host]): F[Unit] =
     for {
+
+      versionHash <- BuildInfo.version.hashF
```

Comment:
We could possibly calculate it once and then reuse? 🤔

Reply:
where/when?

Reply:
I'm not in favor of tiny optimizations

Reply:
Maybe pass it to make as a Fiber? 🤔

Reply:
isn't it overcomplicated then?

Reply:
Or as Tomek suggests we could calculate it in the TessellationIOApp? 🤔

Reply:
Maybe it's not worth it after all :)

Reply:
My suggestion was to not cache it. It's a rare event that someone is joining.

Reply:
Yeah, so not worth the additional code/obfuscation.